src/offline/generate-address.ts

```diff
@@ -1,19 +1,39 @@
-import keypairs = require('ripple-keypairs')
-import * as common from '../common'
-const {errors, validate} = common
+import {classicAddressToXAddress} from 'ripple-address-codec'
+import * as keypairs from 'ripple-keypairs'
+import {errors, validate} from '../common'
 
 export type GeneratedAddress = {
-  secret: string,
-  address: string
+  xAddress: string,
+  classicAddress: string,
+  address: string, // @deprecated Use `classicAddress` instead.
+  secret: string
 }
 
-function generateAddressAPI(options?: any): GeneratedAddress {
+export interface GenerateAddressOptions {
+  // The entropy to use to generate the seed.
+  entropy?: Uint8Array,
+
+  // The digital signature algorithm to generate an address for. Can be `ecdsa-secp256k1` (default) or `ed25519`.
+  algorithm?: 'ecdsa-secp256k1' | 'ed25519',
+
+  // Specifies whether the address is intended for use on a test network such as Testnet or Devnet.
+  // If `true`, the address should only be used for testing, and will start with `T`.
+  // If `false` (default), the address should only be used on mainnet, and will start with `X`.
+  test?: boolean
+}
+
+function generateAddressAPI(options?: GenerateAddressOptions): GeneratedAddress {
   validate.generateAddress({options})
   try {
     const secret = keypairs.generateSeed(options)
     const keypair = keypairs.deriveKeypair(secret)
-    const address = keypairs.deriveAddress(keypair.publicKey)
-    return {secret, address}
+    const classicAddress = keypairs.deriveAddress(keypair.publicKey)
+    return {
+      xAddress: classicAddressToXAddress(classicAddress, false, options && options.test),
```

Comment:
Consider whether you want to return classicAddresses here? If we want to push folks towards X-Address maybe we leave this function as is, mark as deprecated, and add a new API for X-address only. 

Pushing more options onto clients lets them be lazy about upgrading and make the programming model more confusing.

Reply:
Good idea -- done